src/HypothesisTests.jl

```diff
@@ -68,35 +68,54 @@ function check_alpha(alpha::Float64)
     end
 end
 
+# Utility to get an optional field (e.g., :tail and :alpha)
+getfield(value::HypothesisTest, name::Symbol, default::Any) =
```

Comment:
Removing the "get_" prefix, I simply called the functions `tail` and `alpha`. By default, the tail is `:undefined`, which is the value returned by `default_tail` if not overloaded.

I feel that the method `default_tail` is kinda superseded by this implementation of the `tail` method. For all tests that always use their default tail, the `tail` method returns that default, just like `default_tail` does. For t-tests, which can be configured to use another tail, that other tail is returned (or the default, if no other tail is specified).

`default_tail` can still be used (and behaves just like `tail`) but it emits a deprecation warning.